aea/helpers/dialogue/base.py

```diff
@@ -656,6 +688,31 @@ def new_self_initiated_dialogue_reference(self) -> Tuple[str, str]:
         """
         return str(self._dialogue_nonce + 1), ""
 
+    def create(self, counterparty, performative, **kwargs,) -> Tuple[Message, Dialogue]:
+        initial_message = self._message_class(
+            dialogue_reference=("", ""),
+            message_id=1,
+            target=0,
+            performative=performative,
+            **kwargs,
+        )
+        initial_message.counterparty = counterparty
+
+        dialogue = self._create_self_initiated(
+            dialogue_opponent_addr=counterparty,
+            role=self.role_from_first_message(initial_message),
+        )
+        initial_message.set(
+            "dialogue_reference", dialogue.dialogue_label.dialogue_reference
+        )
+
+        successfully_updated = dialogue.update(initial_message)
+        assert (
+            successfully_updated
+        ), "Cannot create the a dialogue with the specified performative and contents."
```

Comment:
This is a problem in code; so it should throw rather than being handled.

Reply:
Agreed

Reply:
See above. Refactoring errors to be done for the next release as it would break the interface.

Reply:
create is a new method so you can raise here and ensure it only returns on success